### PR TITLE
[IIO] Added support for other/tensors @open sesame 4/18 17:26

### PR DESCRIPTION
**Pull request by kparichay**
Added support for other/tensors when merging of channels
with corresponding test cases

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Comment by kparichay** [comment]
Rebased

File: tests/nnstreamer_source/unittest_src_iio.cpp [hunk 1]
```diff
@@ -1434,6 +1440,36 @@ TEST (test_tensor_src_iio, data_verify_freq_generic_type)
...
+  ASSERT_NE (src_pad, nullptr);
+  caps = gst_pad_get_current_caps (src_pad);
+  ASSERT_NE (caps, nullptr);
+  structure = gst_caps_get_structure (caps, 0);
+  ASSERT_NE (structure, nullptr);
+
+  /** Default has merge channels enabled */
+  EXPECT_STREQ (gst_structure_get_name (structure), "other/tensors");
+  EXPECT_EQ (gst_tensors_config_from_structure (&config, structure), TRUE);
+  EXPECT_EQ (config.rate_n, samp_freq);
```
Comment (wooksong) [inline tests/nnstreamer_source/unittest_src_iio.cpp]: I wonder that gst_object_unref (src_iio_pipeline) guarantee us unreffering those src_pad, caps, and structure included in src_iio_pipleine. If not, could we do get_object_unref() for variables, src_pad, caps, and structure, here? It seems that there are no code using those variables  after this line. Near the line for gst_object_unref (src_iio_pipeline) looks also good to me.

Comment (kparichay) [reply tests/nnstreamer_source/unittest_src_iio.cpp]: Yes, gst_object_unref (src_iio_pipeline) would only free its resources along with added elements in the pipeline, if they are referenced. Added unref for bin, pad, caps.
Structure need not be unref as per documentation.

**Comment by kparichay** [comment]
V2:
Added unref for objects in tests
Rebased

**Pull request merged**